src/main/shell-session/shell-session.ts

```diff
@@ -61,9 +60,15 @@ export abstract class ShellSession {
   }
 
   open(shell: string, args: string[], env: Record<string, any>): void {
+    let cwd = env.HOME;
+
+    if (this.cwd && fs.existsSync(this.cwd)) {
+      cwd = this.cwd;
+    }
```

Comment:
Nit, this could be 
```suggestion
    const cwd = (this.cwd && await fse.pathExists(this.cwd)) 
    	? this.cwd 
    	: env.HOME;
```

Also, given that both child classes have an `async open` function. I think it would be fine to change this function to async too. Don't want to use blocking FS calls within an async context.

Reply:
Agreed, this is an oversight on my part about the async logic.

And I also prefer the ternary. Will update.